index.js

```diff
@@ -373,8 +373,19 @@ function getQueue(view) {
 }
 
 function updateView(view) {
+  view.sourceDB.on('error', onSourceDbError);
```

Comment:
I think this might be the wrong way to approach this. You might want to look at `tryCode`, where we try to call the map function and then emit an error otherwise. Probably we should change that, instead of adding/removing listeners.